OHIF/Viewers
# feat(grid): remove viewportIndex and only rely on viewportId

## Context
During the development of OHIF, the reliance on viewport index has proven to be problematic. This issue has seen in several ways:

- Inconsistent Viewport Indexing with Layout Changes: A change from a 2x2 to a 3x3 grid alters the viewport index, causing unnecessary re-rendering, even though the display sets remain the same. This has resulted in a lack of optimization in grid rendering during stage and layout changes.

- Viewport Movement Problems: for moving a viewport (similar to moving a person's window in Zoom, which we have in plan to support in future), the change in viewport index forces React to re-render the whole grid (as right now we have the key as viewportIndex). While the element itself hasn't changed, the viewport index change causes a problematic full re-render.

## Changes Made
- Removed Viewport Index: All occurrences of the viewport index have been replaced with viewportId. This prevents the above issues related to the inconsistency of the viewport index.
- Changed Viewport Grid State: The viewport grid state has been modified from an array to a map, using viewport IDs as keys. This provides a more stable mapping of viewports.

## Migration Guide
You should not rely on viewportIndex and activeViewportIndex anymore. 

Instead try 

```
const { activeViewportId, viewports } = viewportGridService.getState() OR useViewportGrid()
```

`Viewports` also has been changed from an array to a Map 


```
viewports: new Map(
    Object.entries({
      default: {
        viewportId: 'default',
        displaySetInstanceUIDs: [],
        viewportOptions: {
          viewportId: 'default',
        },
        displaySetSelectors: [],
        displaySetOptions: [{}],
        x: 0, // left
        y: 0, // top
        width: 100,
        height: 100,
        viewportLabel: null,
      },
    })
  ),

```

So you need to `viewports.get(viewportId` instead of `viewports[viewportIndex]`

## Testing
The implemented changes have been tested, and the following features are working as expected:

- Rendering a default viewport
- Rendering a TMTV
- Rendering segmentation and RT struct
- Enabling double-click on a thumbnail
- Changing layout, going MPR, adding measurements
- Toggle One Up works
- Viewport Labels (ABCD) works
- Presentation States is preserved
- Change from 2x2 to 3x3 works


Cheers
